tests/schedule/test_regular_schedule_strategy.py

```diff
@@ -0,0 +1,46 @@
+from src.schedule.regular_schedule_strategy import RegularScheduleStrategy
+
+
+def test_not_spawning_before_start_tick():
```

Comment:
I like you tests 💚 You covered everything, I think 👍 
If you want to improve them, you could apply [parameterization](https://docs.pytest.org/en/6.2.x/parametrize.html) for example. This reduces magic numbers in tests too :)